engine/src/test/groovy/graphql/nadel/e2e/NadelE2ETest_SharedNamespaceWIP.groovy

```diff
@@ -0,0 +1,409 @@
+            type Query {
+              issue: IssueQuery
+            }
+            
+            type IssueQuery {
+              search: SearchResult
+            }
+            
+            type SearchResult {
+              id: ID
+              count: Int
+            }  
+        ''')
+        def underlyingSchema3 = typeDefinitions('''
+            type Query {
+              page: PageQuery
+            }
+            
+            type PageQuery {
+              getIssue: IssuePage
+            }
+            
+            type IssuePage {
+              id: ID
+              pageText: String
+            }
+        ''')
+
+        def query = '''
+            { 
+              issue {
+                getIssue {
+                  text
+                }
+                
+                search {
+                  count
+                }
+              }
+
+              page {
+                getIssue {
+                  pageText
+                }
+              }
+            }
+        '''
+        ServiceExecution delegatedExecution1 = Mock(ServiceExecution)
+        ServiceExecution delegatedExecution2 = Mock(ServiceExecution)
+        ServiceExecution delegatedExecution3 = Mock(ServiceExecution)
+
+        ServiceExecutionFactory serviceFactory = TestUtil.serviceFactory([
+                Issues     : new Tuple2(delegatedExecution1, underlyingSchema1),
+                IssueSearch: new Tuple2(delegatedExecution2, underlyingSchema2),
+                Pages      : new Tuple2(delegatedExecution3, underlyingSchema3)
+        ])
+
+        given:
+        Nadel nadel = newNadel()
+                .dsl(nsdl)
+                .serviceExecutionFactory(serviceFactory)
+                .build()
+        NadelExecutionInput nadelExecutionInput = newNadelExecutionInput()
+                .query(query)
+                .build()
+        def data1 = [issue: [getIssue: [text: "Foo"]]]
+        def data2 = [issue: [search: [count: 1]]]
+        def data3 = [page: [getIssue: [pageText: "Bar"]]]
+        ServiceExecutionResult delegatedExecutionResult1 = new ServiceExecutionResult(data1)
+        ServiceExecutionResult delegatedExecutionResult2 = new ServiceExecutionResult(data2)
+        ServiceExecutionResult delegatedExecutionResult3 = new ServiceExecutionResult(data3)
+        when:
+        def result = nadel.execute(nadelExecutionInput)
+
+        then:
+        1 * delegatedExecution1.execute(_) >> completedFuture(delegatedExecutionResult1)
+        1 * delegatedExecution2.execute(_) >> completedFuture(delegatedExecutionResult2)
+        1 * delegatedExecution3.execute(_) >> completedFuture(delegatedExecutionResult3)
+        def er = result.join()
+        er.data == [
+                issue: [
+                        getIssue: [text: "Foo"],
+                        search  : [count: 1]
+                ],
+                page : [
+                        getIssue: [pageText: "Bar"]
+                ]
+        ]
+    }
+}
```

Comment:
Is it worth adding more complex tests that include renames and hydration?

Reply:
we will have to do more work to support renames and hydrations, will be done in the follow up PR